ui/app/actions.js

```diff
@@ -674,21 +674,25 @@ function markAccountsFound() {
 // default rpc target refers to localhost:8545 in this instance.
 function setDefaultRpcTarget (rpcList) {
   log.debug(`background.setDefaultRpcTarget`)
-  background.setDefaultRpc()
-  return {
-    type: actions.SET_RPC_TARGET,
-    value: 'http://localhost:8545',
+  return (dispatch) => {
+    background.setDefaultRpc((err, result) => {
+      if (err) {
+        console.error(err)
+        return dispatch(self.displayWarning('Had a problem changing networks.'))
+      }
+    })
   }
 }
 
 function setRpcTarget (newRpc) {
+  log.debug(`background.setRpcTarget`)
   return (dispatch) => {
-    log.debug(`background.setRpcTarget`)
-    background.setCustomRpc(newRpc)
-    return {
-      type: actions.SET_RPC_TARGET,
-      value: newRpc,
-    }
+    background.setCustomRpc(newRpc, (err, result) => {
+      if (err) {
+        console.err(err)
```

Comment:
Should be console.error
http://mdn.beonex.com/en/DOM/console.error.html

Or even better, use the new global 'log' object, so if we ever add more tooling to our custom logs we are already set up.

It's loglevel.